controllers/vspheremachine_controller.go

```diff
@@ -414,6 +414,9 @@ func (r machineReconciler) reconcileNormalPre7(ctx *context.MachineContext) (run
 		if vm.Spec.ResourcePool == "" {
 			vm.Spec.ResourcePool = vsphereCloudConfig.ResourcePool
 		}
+
+		vm.Spec.EnableSecureVCenterTransport = ctx.VSphereCluster.Spec.CloudProviderConfiguration.Global.Insecure
```

Comment:
if nothing is set on `VSphereCluster` this should default to true for backward-compatibility

Reply:
Yeah, since `Insecure` is an optional field we should probably make it a pointer

Reply:
Hmm that would be breaking change though ? let's summarize

- if vspherecluster insecure is true adding this code shouldn't break anyone
- if vspherecluster insecure is set to false adding this code is still okay, as the CPI wouldn't be working if there was an issue with `insecure: false`, so vm provisioning should be the same

so no need to change something here

Reply:
right, I pushed some changes PTAL